contracts/Swaps.vy

```diff
@@ -303,9 +315,8 @@ def get_best_rate(_from: address, _to: address, _amount: uint256) -> (address, u
             pool: address = Registry(registry).find_pool_for_coins(_from, _to, i)
             if pool == ZERO_ADDRESS:
                 break
-
-            dy: uint256 = self._get_exchange_amount(registry, pool, _from, _to, _amount)
-            if dy > max_dy:
+            dy: uint256 = self._get_exchange_amount(self.registry, pool, _from, _to, _amount)
+            if dy > max_dy and not pool in _exclude_pools:
```

Comment:
The check for `pool in _exclude_pools` should happen prior to `_get_exchange_amount`.